### PR TITLE
Using github-app-installation-token pacakge + esbuild 

### PR DESCRIPTION
**Pull request by gagoar**
This PR changes a couple of things:

1) it uses [github-app-installation-token](https://github.com/gagoar/github-app-installation-token)
2) It changes ncc for esbuild due to the known issues with [is-plain-object](https://github.com/octokit/graphql.js/issues/195#issuecomment-695368139) and [octokit/rest](https://github.com/octokit/rest.js/issues/1929)
3) changed fetch-mock in favor of nock for convenience.

### CODE REVIEW

**Comment by gagoar** [comment]
some steps are failing because secrets aren't in my repo.

File: __tests__/main.test.ts [hunk 1]
```diff
@@ -35,28 +47,36 @@ ZcJjRIt8w8g/s4X6MhKasBYm9s3owALzCuJjGzUKcDHiO2DKu1xXAb0SzRcTzUCn
...
+
+  getInput.mockImplementation((key: keyof typeof mockedInput): string =>
+    String(mockedInput[key])
   )
 
-  expect(token).toEqual('secret-installation-token123')
-  expect(mock.done()).toBe(true)
+  try {
+    await run()
+    debugger
```
Comment (jnwng) [inline __tests__/main.test.ts]:
```suggestion

```

File: package.json [hunk 2]
```diff
@@ -26,25 +25,23 @@
   "license": "MIT",
   "dependencies": {
     "@actions/core": "^1.2.0",
-    "@octokit/auth-app": "^2.4.7",
-    "@octokit/rest": "^18.0.0",
-    "fetch-mock": "^9.10.3"
+    "@types/nock": "^11.1.0",
+    "esbuild": "^0.8.3",
```
Comment (jnwng) [inline package.json]: these should be devDependencies

**Review by jnwng — CHANGES_REQUESTED** [review]

**Comment by gagoar** [comment]
closed in favour of #5

**Pull request closed without merging**